### ISSUE TITLE
nomnom package is no longer maintained

### ISSUE THREAD

**Issue by alliance-loop** [kach/nearley#380]
The `nomnom` dependency is no longer maintained (see https://github.com/harthur/nomnom#deprecation-notice) and an alternative needs to be found

**Comment by xakep139** [comment]
Unfortunately all related issues are currently closed: https://github.com/kach/nearley/issues/369, https://github.com/kach/nearley/issues/364, https://github.com/kach/nearley/issues/353, https://github.com/kach/nearley/issues/352 and PR https://github.com/kach/nearley/pull/284

**Comment by golgistudio** [comment]
And the license for nomnom is 'None' - makes it hard for those organizations that do license checks on the package sub dependencies.  (this package is used by rst-selector-parser which is used by enzyme)

Commander has been mentioned as an alternative.
https://www.npmjs.com/package/commander

**Comment by honzajavorek** [comment]
There is even a PR to solve this, but it was rejected. I commented with more reasoning on why the transition should happen: https://github.com/kach/nearley/pull/284#issuecomment-417690840

**Comment by nihonjinrxs** [comment]
Latest version of `nomnom` seems to be 1.8.1, but also deprecated and not installable. The message on the npm page, and in the output when you try to install is:
```
Package no longer supported. Contact support@npmjs.com for more info.
```
(ref. https://www.npmjs.com/package/nomnom)

Because of this, even if you want to keep using `nomnom`, relying on install via `package.json` for this dependency seems unwise.

I can think of a few paths forward. (There may be more that I'm not thinking of at the moment.) Hopefully one of these is suitable.

* Choose a different CLI parsing library. The original author of `nomnom` has suggested [`commander`](https://www.npmjs.com/package/commander) (ref. [`harthur/nomnom`](https://github.com/harthur/nomnom#deprecation-notice)).
* Use one of the several forks of `nomnom` published to npm, for example [`@gerhobbelt/nomnom`](https://www.npmjs.com/package/@gerhobbelt/nomnom).
* Contact the original author of `nomnom`, and take over ownership and maintenance, then continue using it.
* Pull `nomnom` into this codebase and maintain it as part of `nearley`.
* Rewrite the CLI to use `nearley` to parse its own command-line options.

(Although most do not address the dependency licensing issue #396 referenced above, which will remain if `nomnom` is kept as a dependency.)

Are any of these something that might be palatable for removing this dependency issue?

**Comment by tjvr** [comment]
@kach 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kach** [comment]
Hopefully fixed in 2.16.0.
